Code/Source/RobotImporter/URDF/RobotImporter.cpp

```diff
@@ -42,21 +45,28 @@ namespace ROS2
 
     void RobotImporter::CheckIfAssetsWereLoadedAndCreatePrefab(std::function<void()> importFinishedCb)
     {
-        AZ_Assert(m_prefabMaker, "Prefab maker is not initialized");
-        if (m_isProcessingAssets)
+        if (m_prefabMaker)
         {
-            return;
+            if (m_isProcessingAssets)
+            {
+                return;
+            }
+            auto outcome = m_prefabMaker->CreatePrefabFromURDF();
+            if (!outcome)
+            {
+                auto errorMessage = AZStd::string::format("Importing robot definition failed with error: %s", outcome.GetError().c_str());
+                m_logger(LogLevel::Error, errorMessage);
+                importFinishedCb();
+                return;
+            }
+            m_logger(LogLevel::Info, AZStd::string::format("Imported %s", m_prefabMaker->GetPrefabPath().c_str()));
+            importFinishedCb();
         }
-
-        auto outcome = m_prefabMaker->CreatePrefabFromURDF();
-        if (!outcome)
+        else if (m_LoadingURDFFailed)
         {
-            auto errorMessage = AZStd::string::format("Importing robot definition failed with error: %s", outcome.GetError().c_str());
-            m_logger(LogLevel::Error, errorMessage);
             importFinishedCb();
             return;
         }
-        m_logger(LogLevel::Info, AZStd::string::format("Imported %s", m_prefabMaker->GetPrefabPath().c_str()));
-        importFinishedCb();
+        AZ_Assert(m_prefabMaker, "Prefab maker is not initialized");
```

Comment:
I believe this is an unnecessary assert. Move it to the top or remove it.

Reply:
I do not agree - we need to assert when URDF is not failed, but m_prefab optional is empty. I changed simplified logic in this function to make this assert more relevant. Hope you like it more now.

Reply:
I was thinking about the rule in general, that we should assert that "something" is "correct" before we start using it in any way. 

But yes, I see the reason behind that implementation. In this case, I'm ok with it.